core/src/test/scala/org/apache/spark/util/SizeEstimatorSuite.scala

```diff
@@ -135,5 +160,20 @@ class SizeEstimatorSuite
     assertResult(64)(SizeEstimator.estimate(DummyString("a")))
     assertResult(64)(SizeEstimator.estimate(DummyString("ab")))
     assertResult(72)(SizeEstimator.estimate(DummyString("abcdefgh")))
+    
```

Comment:
I'm worried that setting the system props above like `os.arch` is going to pollute the JVM state for other tests. It should be restored.